Code/Framework/AzCore/AzCore/Name/NameDictionary.cpp

```diff
@@ -81,10 +95,16 @@ namespace AZ
     }
     
     NameDictionary::NameDictionary()
-    {}
+    {
+        m_deferredHead = s_deferredHead;
+        LoadDeferredNames(AZ::Name::GetDeferredHead());
+        LoadDeferredNames(m_deferredHead);
```

Comment:
You should be able to remove the s_deferredHead. There is no need to unlink the Name entries for the current memory space into a linked list, because it already exist in the Name deferred head.